src/components/calcite-color-picker/calcite-color-picker.tsx

```diff
@@ -928,143 +1025,65 @@ export class CalciteColorPicker {
 
     this.drawColorFieldAndSlider();
 
-    const yWithin = (y: number): "color-field" | "slider" | "none" => {
-      const {
-        dimensions: {
-          colorField: { height: colorFieldHeight },
-          slider: { height: sliderHeight }
-        }
-      } = this;
-
-      if (y <= colorFieldHeight) {
-        return "color-field";
-      }
-
-      if (y <= colorFieldHeight + sliderHeight) {
-        return "slider";
-      }
-
-      return "none";
-    };
+    let activeThumbX: number;
+    let activeThumbY: number;
 
-    canvas.addEventListener("mousedown", ({ offsetX, offsetY }) => {
-      const region = yWithin(offsetY);
+    canvas.addEventListener("mousedown", (event) => {
+      const { offsetX, offsetY } = event;
+      activeThumbX = offsetX;
+      activeThumbY = offsetY;
+      const region = this.getCanvasRegion(offsetY);
 
       if (region === "color-field") {
         this.hueThumbState = "drag";
         this.captureColorFieldColor(offsetX, offsetY);
       } else if (region === "slider") {
         this.sliderThumbState = "drag";
-        captureHueSliderColor(offsetX);
+        this.captureHueSliderColor(offsetX);
       }
-    });
-
-    canvas.addEventListener("mouseout", () => {
-      this.hueThumbState = "idle";
-      this.sliderThumbState = "idle";
 
-      this.drawColorFieldAndSlider();
-    });
-
-    canvas.addEventListener("mouseup", () => {
-      this.hueThumbState = "hover";
-      this.sliderThumbState = "hover";
+      // prevent text selection outside of color field & slider area
+      event.preventDefault();
 
-      this.drawColorFieldAndSlider();
+      document.addEventListener(
```

Comment:
Are these document events being removed at some point? I'm not seeing anything.

Should maybe these events be added to `getRootNode()` instead of document?

Reply:
> Are these document events being removed at some point? I'm not seeing anything.

🤦🏼  Great catch. Will fix!

> Should maybe these events be added to getRootNode() instead of document?

I need `document` to track these events across the page.